test/Atmos/EDMF/edmf_kernels.jl

```diff
@@ -882,12 +841,26 @@ function precompute(::EDMF, bl, args, ts, ::Flux{FirstOrder})
     env = environment_vars(state, aux, n_updrafts(bl.turbconv))
     ρa_up = compute_ρa_up(bl, state, aux)
     up = state.turbconv.updraft
+    gm = state
     N_up = n_updrafts(bl.turbconv)
+    ρ_inv = 1 / gm.ρ
     w_up = vuntuple(N_up) do i
         fix_void_up(ρa_up[i], up[i].ρaw / ρa_up[i])
     end
 
-    return (; env, ρa_up, w_up, fix_void_up)
+    θ_liq_up = vuntuple(N_up) do i
+        fix_void_up(up[i].ρa, up[i].ρaθ_liq / up[i].ρa, liquid_ice_pottemp(ts))
+    end
+    a_up = vuntuple(N_up) do i
+        fix_void_up(up[i].ρa, up[i].ρa * ρ_inv)
+    end
+    if !(bl.moisture isa DryModel)
+        q_tot_up = vuntuple(N_up) do i
+            fix_void_up(up[i].ρa, up[i].ρaq_tot / up[i].ρa, gm.moisture.ρq_tot)
+        end
+    end
+
+    return (; env, a_up, q_tot_up, ρa_up, θ_liq_up, w_up, fix_void_up)
```

Comment:
I think `fix_void_up` is not needed in the return, I probably left this here in some PR.